api/models/student.py

```diff
@@ -36,3 +38,14 @@ def meets_requirement(self, requirement: ProjectRequirement) -> bool:
             else:  # default case is RequirementOperator.EXACTLY
                 is_met |= value == requirement.value
         return is_met
+
+    def validate(self):
```

Comment:
Like this is something that the validator class would do, I wouldn't put it here imo